tests/test_odp_event_manager.py

```diff
@@ -0,0 +1,273 @@
+# Copyright 2022, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import queue
+from unittest import mock
+import uuid
+
+from optimizely.odp.odp_event import OdpEvent
+from optimizely.odp.odp_event_manager import OdpEventManager
+from optimizely.odp.odp_config import OdpConfig
+from . import base
+from optimizely.version import __version__
+
+
+class OdpEventManagerTest(base.BaseTest):
+    user_key = "vuid"
+    user_value = "test-user-value"
+    api_key = "test-api-key"
+    api_host = "https://test-host.com"
+    test_uuid = str(uuid.uuid4())
+    odp_config = OdpConfig(api_key, api_host)
+
+    events = [
+        {
+            "type": "t1",
+            "action": "a1",
+            "identifiers": {"id-key-1": "id-value-1"},
+            "data": {"key-1": "value1"}
+        },
+        {
+            "type": "t2",
+            "action": "a2",
+            "identifiers": {"id-key-2": "id-value-2"},
+            "data": {"key-2": "value2"}
+        }
+    ]
+
+    processed_events = [
+        {
+            "type": "t1",
+            "action": "a1",
+            "identifiers": {"id-key-1": "id-value-1"},
+            "data": {
+                "idempotence_id": test_uuid,
+                "data_source_type": "sdk",
+                "data_source": "python-sdk",
+                "data_source_version": __version__,
+                "key-1": "value1"
```

Comment:
Can we have a test case that custom data overwrites the default data like {"data_source": "my-app"}